### PR TITLE
Update upgrade provider workflow

### PR DESCRIPTION
**Pull request by aq17**
Update auto upgrade provider workflow. Tested successfully on pulumi-pagerduty:
https://github.com/pulumi/pulumi-pagerduty/blob/aq17-upgrade-provider-wkflw/.github/workflows/upgrade-provider.yml
https://github.com/pulumi/pulumi-pagerduty/pull/184

### CODE REVIEW

**Comment by aq17** [comment]
Wait for https://github.com/pulumi/upgrade-provider/pull/46 to land

File: provider-ci/providers/aiven/repo/.github/workflows/upgrade-provider.yml [hunk 1]
```diff
@@ -1,27 +1,65 @@
 # WARNING: This file is autogenerated - changes will be overwritten if not made via https://github.com/pulumi/ci-mgmt
 
 env:
+  GH_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
   GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 jobs:
-  update_bridge:
```
Comment (iwahbe) [inline provider-ci/providers/aiven/repo/.github/workflows/upgrade-provider.yml:6]: Does this supersede the `update_bridge` workflow?

**Review by iwahbe — APPROVED** [review]
Beyond my question, this looks good to me.

File: provider-ci/providers/aiven/repo/.github/workflows/upgrade-provider.yml (hunk 1 above)
Comment (aq17) [reply provider-ci/providers/aiven/repo/.github/workflows/upgrade-provider.yml:6]: No, it's the local name of the job. I think it was left as `update_bridge` from when I copy/ pasted that workflow as a starting point.

File: provider-ci/providers/aiven/repo/.github/workflows/upgrade-provider.yml [hunk 2]
```diff
@@ -9,54 +9,16 @@ jobs:
     name: upgrade-provider
     runs-on: ubuntu-latest
     steps:
-    - name: Install Go
-      uses: actions/setup-go@v3
+    - name: Call upgrade provider action
+      uses: pulumi/pulumi-upgrade-provider-action@v0.0.4
```
Comment (aq17) [inline provider-ci/providers/aiven/repo/.github/workflows/upgrade-provider.yml:13]: Wait til we finalize https://github.com/pulumi/pulumi-upgrade-provider-action/blob/main/action.yml and release an official v1.0.0

**Pull request merged**